ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterFactory.java

```diff
@@ -59,7 +62,7 @@
    * @throws IllegalArgumentException if any of the arguments are null.
    */
   public NonBlockingRouterFactory(VerifiableProperties verifiableProperties, ClusterMap clusterMap,
-      NotificationSystem notificationSystem, SSLFactory sslFactory) throws GeneralSecurityException {
+      NotificationSystem notificationSystem, SSLFactory sslFactory) throws Exception {
```

Comment:
Why does this now throw `Exception`?

Reply:
since we instantiate KMS and crypto, Utils.getObj() throws ClassNotFoundException, InstantiationException, IllegalAccessException, NoSuchMethodException,
             InvocationTargetException